packages/shared/src/utils/parse-filter.ts

```diff
@@ -2,80 +2,66 @@ import { REGEX_BETWEEN_PARENS } from '../constants';
+		} else if (['_in', '_nin', '_between', '_nbetween'].includes(String(key))) {
+			result[key] = toArray(value).flatMap((value) => parseFilterValue(value, accountability, context));
+		} else if (String(key).startsWith('_')) {
+			result[key] = parseFilterValue(value, accountability, context);
+		} else {
+			result[key] = parseFilter(value, accountability, context);
 		}
+		return result as Filter;
+	}, {} as any);
+}
 
-		if (isDynamicVariable(val)) {
-			if (val.startsWith('$NOW')) {
-				if (val.includes('(') && val.includes(')')) {
-					const adjustment = val.match(REGEX_BETWEEN_PARENS)?.[1];
-					if (!adjustment) return new Date();
-					return adjustDate(new Date(), adjustment);
-				}
-
-				return new Date();
-			}
-
-			if (val.startsWith('$CURRENT_USER')) {
-				if (val === '$CURRENT_USER') return accountability?.user ?? null;
-				return get(context, val, null);
-			}
+function parseFilterValue(value: any, accountability: Accountability | null, context: ParseFilterContext) {
+	if (value === 'true') return true;
+	if (value === 'false') return false;
+	if (value === 'null' || value === 'NULL') return null;
+	if (isDynamicVariable(value)) return parseDynamicVariable(value, accountability, context);
+	return value;
+}
 
-			if (val.startsWith('$CURRENT_ROLE')) {
-				if (val === '$CURRENT_ROLE') return accountability?.role ?? null;
-				return get(context, val, null);
-			}
+function parseDynamicVariable(value: any, accountability: Accountability | null, context: ParseFilterContext) {
+	if (value.startsWith('$NOW')) {
+		if (value.includes('(') && value.includes(')')) {
+			const adjustment = value.match(REGEX_BETWEEN_PARENS)?.[1];
+			if (!adjustment) return new Date();
+			return adjustDate(new Date(), adjustment);
 		}
 
-		return val;
+		return new Date();
 	}
-}
 
-function get(obj: Record<string, any> | any[], path: string, defaultValue: any) {
-	const pathParts = path.split('.');
-	let val = obj;
-
-	while (pathParts.length) {
-		const key = pathParts.shift();
-
-		if (key) {
-			val = processLevel(val, key);
-		}
+	if (value.startsWith('$CURRENT_USER')) {
+		if (value === '$CURRENT_USER') return accountability?.user ?? null;
+		return get(context, value, null);
 	}
 
-	return val || defaultValue;
-
-	function processLevel(value: Record<string, any> | any[], key: string) {
-		if (Array.isArray(value)) {
-			return value.map((subVal) => subVal[key]);
-		} else if (value && typeof value === 'object') {
-			return value[key];
-		}
+	if (value.startsWith('$CURRENT_ROLE')) {
+		if (value === '$CURRENT_ROLE') return accountability?.role ?? null;
+		return get(context, value, null);
 	}
 }
+
+function get(object: Record<string, any> | any[], path: string, defaultValue: any): any {
+	const [key, ...rest] = path.split('.');
+	const result = Array.isArray(object) ? object.map((entry) => entry[key!]) : object?.[key!];
+	return rest.length === 0 ? result ?? defaultValue : get(result, rest.join('.'), defaultValue);
+}
```

Comment:
While this is a very smart implementation, it's way harder to read and follow along what's going on. Is there any practical benefit to this refactor? I'd prefer keeping the older implementation for maintenance reasons if it's only a stylistic change.

Reply:
Is it fine with the change I made or do you want to revert to the previous implementation ?